SPEC/FILES.md

```diff
@@ -709,28 +709,90 @@ ipfs.files.rm('/my/beautiful/directory', { recursive: true }, (err) => {
 
 #### `read`
 
-> Read a file.
+> Read a file into a [`Buffer`][b].
 
 ##### `Go` **WIP**
 
-##### `JavaScript` - ipfs.files.read(path, [options, callback])
+##### `JavaScript` - ipfs.files.read(path, [options], [callback])
 
 Where:
 
-- `path` is the path of the object to read.
+- `path` is the path of the file to read.
 - `options` is an optional Object that might contain the following keys:
   - `offset` is an Integer with the byte offset to begin reading from.
   - `count` is an Integer with the maximum number of bytes to read.
 
-`callback` must follow the `function (err, buf) {}` signature, where `err` is an Error if the operation was not successful and `buf` is a Buffer with the contents of `path`.
+`callback` must follow the `function (err, buf) {}` signature, where `err` is an Error if the operation was not successful and `buf` is a [`Buffer`][b] with the contents of `path`.
 
 If no `callback` is passed, a promise is returned.
 
 **Example:**
 
 ```JavaScript
 ipfs.files.read('/hello-world', (err, buf) => {
-  console.log(buf.toString())
+  console.log(buf.toString('utf8'))
+})
+
+// Hello, World!
+```
+
+#### `readReadableStream`
+
+> Read a file into a [`ReadableStream`][rs].
+
+##### `Go` **WIP**
+
+##### `JavaScript` - ipfs.files.readReadableStream(path, [options], [callback])
```

Comment:
Other stream based functions do not callback/promise, they return a stream. @diasdavid should these also for consistency?

Reply:
That might make for a slightly nicer API

Reply:
Yeah, @alanshaw is right. No need for using callbacks.

Reply:
I've updated the PR to not use callbacks/promises for streams.